src/main/scala/com/databricks/labs/overwatch/env/Database.scala

```diff
@@ -27,18 +27,54 @@ class Database(config: Config) extends SparkSessionWrapper {
   }
 
   /**
-   * register an Overwatch target table in the configured Overwatch deployment
+   * Checks if the table has optimizeWrite enabled.
+   * @param target
+   * @return
+   */
+  private def checkAutoOptimizeWrite(target: PipelineTable): Boolean = {
+    import spark.implicits._
+    val optimiseWrite: Boolean = spark.sql(s"""describe detail '${target.tableLocation}'""")
+      .select("properties").as[Map[String, String]].collect()
+      .headOption.getOrElse(throw new Exception("string"))
+      .getOrElse("delta.autoOptimize.optimizeWrite", "false")
+      .toBoolean
+    optimiseWrite
+  }
+
+
+  /**
+   * register an Overwatch target table in the configured Overwatch deployment.
+   * After release > 7.2.1 : When the tables are dropped ,it will create the table as it was previously created then check whether it has autoOptimize enabled or not.
+   * If AutoOptimize is enables then alter the table and set the property of autoOptimizeWrite to true.
    *
    * @param target Pipeline table (i.e. target) as per the Overwatch deployed config
    */
   def registerTarget(target: PipelineTable): Unit = {
     if (!target.exists(catalogValidation = true) && target.exists(pathValidation = true)) {
-      val createStatement = s"create table if not exists ${target.tableFullName} " +
-        s"USING DELTA location '${target.tableLocation}'"
-      val logMessage = s"CREATING TABLE: ${target.tableFullName} at ${target.tableLocation}\n$createStatement\n\n"
-      logger.log(Level.INFO, logMessage)
-      if (config.debugFlag) println(logMessage)
-      spark.sql(createStatement)
+      if (target.autoOptimize) {//Checks if the target table should be autoOptimize or not.
+        val targetPropertiesAccurate = checkAutoOptimizeWrite(target)
+        val createStatement = s"create table if not exists ${target.tableFullName} " +
+          s"USING DELTA location '${target.tableLocation}' " +
+          s"TBLPROPERTIES (delta.autoOptimize.optimizeWrite = ${targetPropertiesAccurate},delta.autoOptimize.autoCompact=false)"
+        val logMessage = s"CREATING TABLE: ${target.tableFullName} at ${target.tableLocation}\n$createStatement\n\n"
+        logger.log(Level.INFO, logMessage)
+        if (config.debugFlag) println(logMessage)
+        spark.sql(createStatement) //Creating the table as it was previously created.
+        if (!targetPropertiesAccurate) {
+          val alterStatement = s"""alter table ${target.tableFullName} set TBLPROPERTIES (delta.autoOptimize.optimizeWrite = true)"""
```

Comment:
Since you're inside an if block of `!target.exists(catalogValidation = true)` then this will never work because the table does not exist once you get to this block of code since alter table tableName will result in Table Does not Exist.